plugins/kubernetes-backend/src/service/KubernetesBuilder.ts

```diff
@@ -273,6 +279,12 @@ export class KubernetesBuilder {
       });
     });
 
+    router.get('/proxy/:encodedQuery', this.makeProxyRequest.bind(this));
```

Comment:
@Rugvip & @jamieklassen Sorry if I'm missing something but doesn't this look like the builder pattern is not implemented correctly? it looks to me like the built router will need a reference to the builder instance, I thought the builder should be used only to build the router then be discarded, I would instead move `makeProxyRequest` to the Proxy instance and rely on a scoped instance just like the `router` constant in line :243, what do you think?

Reply:
It feels like the builder is more than just a builder, it also holds information and implementation details, I thought this was supposed to build a `KubernetesClientProvider` or one of the other services, but I think at this point the `KubernetesBuilder` is both the builder and the instance that holds some of the other instances the router needs to work...